bitfinex/websockets/client.py

```diff
@@ -105,9 +106,36 @@ def _start_socket(self, id_, payload, callback):
         factory.reconnect = True
         context_factory = ssl.ClientContextFactory()
         self.factories[id_] = factory
-        self._conns[id_] = connectWS(factory, context_factory)
+
+        #reconnect treatment
+        if not self.is_alive():
+            self._conns[id_] = connectWS(factory, context_factory)
+        else:
+            # https://twistedmatrix.com/documents/12.2.0/core/howto/threading.html
+            # https://bit.ly/2NypJR3
+            reactor.callFromThread(self.add_connection, id_)
+            ## must wait for connectWS called from thread to finish
+            lim_retries = 0
+            while (id_ not in self._conns) or (lim_retries > 20):
+                lim_retries += 1
+                sleep(0.2)
```

Comment:
Number of retries and time between retries should be configurable. 

Are you retrying to connect every 0.2 seconds here? That seems a bit too fast, have you seen this successfully reconnect?

Reply:
0.2 seconds was enough  time in testing for the authenticate to finish, haven't seen a second iteration on this while

Reply:
Ah, ok now i understand. this is just to wait for the connection to be established.